app/src/main/java/com/zeapo/pwdstore/autofill/v2/StructureParser.kt

```diff
@@ -0,0 +1,137 @@
+            val inferredHint = inferHint(node.idEntry)
+            if (inferredHint != null) {
+                hints = arrayOf(inferredHint)
+            }
+        }
+
+        if (hints != null && hints.isNotEmpty()) {
+            val hintsAsList = listOf(*hints)
+
+            node.autofillId?.let {
+                when {
+                    hintsAsList.contains(View.AUTOFILL_HINT_USERNAME) -> result.username.add(it)
+                    hintsAsList.contains(View.AUTOFILL_HINT_EMAIL_ADDRESS) -> result.email.add(it)
+                    hintsAsList.contains(View.AUTOFILL_HINT_PASSWORD) -> result.password.add(it)
+                    !hintsAsList.contains("") && !hintsAsList.contains("off")  -> {
+                        // For now assume every input as fillable (except off and empty hint)
+                        result.password.add(it)
+                    }
+                    else -> {
+                        // Ignored
+                    }
+                }
+            }
+        } else if (node.autofillType == View.AUTOFILL_TYPE_TEXT) {
+            // Attempt to match based on Field Type
+            node.autofillId?.let {
+                when (node.inputType) {
+                    InputType.TYPE_TEXT_VARIATION_WEB_EMAIL_ADDRESS -> result.email.add(it)
+                    InputType.TYPE_TEXT_VARIATION_PASSWORD -> result.password.add(it)
+                    InputType.TYPE_TEXT_VARIATION_WEB_PASSWORD -> result.password.add(it)
+                    else -> {
+                        // Ignored
+                    }
+                }
+
+            }
+        }
+
+        // Finally look for domain names
+        val webDomain = node.webDomain
+        if (webDomain != null) {
+            result.webDomain.add(webDomain)
+        }
+
+        for (i in 0 until node.childCount) {
+            parseViewNode(node.getChildAt(i))
+        }
+
+    }
+
+    // Attempt to infer the AutoFill type from a string
+    private fun inferHint(actualHint: String?): String? {
+        if (actualHint == null) return null
+
+        val hint = actualHint.toLowerCase()
+        if (hint.contains("label") || hint.contains("container")) {
+            return null
+        }
+
+        if (hint.contains("password") || hint.contains("pass")) {
+            return View.AUTOFILL_HINT_PASSWORD
+        }
+
+        if (hint.contains(View.AUTOFILL_HINT_USERNAME) || hint.contains("login") || hint.contains("id") || hint.contains("user name") || hint.contains("identifier")) {
+            return View.AUTOFILL_HINT_USERNAME
+        }
+
+        if (hint.contains(View.AUTOFILL_HINT_EMAIL_ADDRESS) || hint.contains("email")) {
+            return View.AUTOFILL_HINT_EMAIL_ADDRESS
+        }
+
+        if (hint.contains("name")) {
+            return View.AUTOFILL_HINT_NAME
+        }
+
+        return if (hint.contains("phone")) {
+            View.AUTOFILL_HINT_PHONE
+        } else {
+            null
+        }
```

Comment:
Put the conditions for each branch into an array, then use the `any` map operator to return values.